### PR TITLE
aXe 3 upgrade & Support for React 16

### PR DESCRIPTION
**Pull request by marcysutton**
React-axe didn't need much work at all for axe-core 3.0. It took more work to test that it worked properly, so that's where the bulk of my time was spent. I also added a fix for React 16 and verified it still works with prior versions.

### CODE REVIEW

File: example/package.json [hunk 1]
```diff
@@ -10,18 +10,24 @@
   "author": "Dylan Barrell (dylan@barrell.com)",
   "license": "MIT",
   "dependencies": {
-    "axe-core": "^2.6.1",
+    "axe-core": "^3.0.0-alpha.9",
     "browserify": "^16.0.0",
-    "react": "^15.3.0",
-    "react-dom": "^15.3.0"
+    "create-react-class": "^15.6.3",
+    "document-ready-promise": "^3.0.1",
```
Comment (WilcoFiers) [inline example/package.json]: Are you sure about this dependency?

**Comment by WilcoFiers** [comment]
@marcysutton Should we update to React 16?

**Comment by marcysutton** [comment]
@WilcoFiers why wouldn't we update? It was a pretty easy change.

File: example/package.json (hunk 1 above)
Comment (marcysutton) [reply example/package.json]: I removed it. But I needed a new release of axe-core to finish this work

**Pull request merged**